core/src/main/java/org/elasticsearch/common/xcontent/ParseFieldRegistry.java

```diff
@@ -53,6 +53,13 @@ public ParseFieldRegistry(String registryName) {
     /**
      * Register a parser.
      */
+    public void register(T value, String name) {
+        register(value, new ParseField(name));
+    }
+
+    /**
+     * Register a parser.
+     */
```

Comment:
is it necessary to add another way to register queries? At the end of the day it creates the ParseField, you just don't like all those constants?